### PR TITLE
Add Checker to wrap snap-store-compare behavior

### PR DESCRIPTION
**Pull request by redgoat650**
Checker connects the snapshotter, the storer, and the comparer. It
uses the snapshotter to take/restore/delete snapshots on the repo,
uses the comparer to gather metadata before the snapshot and
after the restore, and uses the storer to save metadata for later
lookup when verifying restores.

This PR incorporates changes from
- #40 
- #41 
- #43 
so I would recommend reviewing those first! However, the only change reflected here that isn't already in the other PRs is adding `checker.go`.

### CODE REVIEW

**Comment by redgoat650** [comment]
Note: this PR will be part of a series that will eventually converge to https://github.com/kopia/kopia/compare/master...kastenhq:test-engine . See that comparison for the full context of how and where this change will be used.

File: tests/robustness/checker/checker.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+	"github.com/kopia/kopia/tests/robustness/snapif"
+	"github.com/kopia/kopia/tests/robustness/snapstore"
+)
+
+// Checker is an object that can take snapshots and restore them, performing
+// a validation for data consistency
+type Checker struct {
+	RestoreDir string
+	snap       snapif.Snapshotter
```
Comment (julio-lopez) [inline tests/robustness/checker/checker.go]: If one has not looked in detail at the other packages and type definitions, it is not immediately obvious from the field type or the names what their intended use is.

Regarding the package naming comment in the other PR. While picking a good name is always hard, it'd be good to end up with package and type names that self-describe the fields.

File: tests/robustness/checker/checker.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	}
+}
+
+// GetSnapIDs gets the list of snapshot IDs being tracked by the checker's snapshot store
+func (chk *Checker) GetSnapIDs() []string {
+	return chk.snapStore.GetKeys()
+}
+
+// SnapshotMetadata holds metadata associated with a given snapshot
+type SnapshotMetadata struct {
```
Comment (julio-lopez) [inline tests/robustness/checker/checker.go]: it makes sense to add JSON tags to the struct to follow the convention for JSON objects, in particular field names.

Comment (redgoat650) [reply tests/robustness/checker/checker.go]: added

File: tests/robustness/checker/checker.go (hunk 1 above)
Comment (redgoat650) [reply tests/robustness/checker/checker.go]: renamed the fields

**Comment by redgoat650** [comment]
> 
> 
> I assume that's the change introduced in this PR is the last commit ([f266c31](https://github.com/kastenhq/kopia/commit/f266c31658126b0a5644db160679a1e2dcf2a8ca)). I only looked at that one.
> LG +1 👍

Yep this one had to merge in #40 to get the comparer implementation, which wasn't explicitly in the same train line as the rest

**Comment by julio-lopez** [comment]
LGTM  ✅

**Pull request merged**